src/components/EventDetails/EventDetails.js

```diff
@@ -956,28 +961,53 @@ class EventDetail extends React.Component<Props, State> {
             squarePrimary: true,
           };
 
+          const sendBackBtc = {
+            title: 'Send back',
+            onPress: () => this.sendToBtc(event.from),
+            secondary: true,
+          };
+
+          const sendMoreBtc = {
+            title: 'Send more',
+            onPress: () => this.sendToBtc(event.to),
+            secondary: true,
+          };
+
+
           if (isReferralRewardTransaction) {
             buttons = [];
           } else if (isReceived) {
-            if (isKWAddress(event.from, accounts) && isSWAddress(event.to, accounts)) {
-              buttons = [send, topUpMore];
-            } else if (contactFound) {
-              if (isPending) {
-                buttons = [messageButton, viewOnBlockchainButtonSecondary];
+            if (isFromKWToSW) {
+              buttons = [sendFromSW, topUpMore];
+            } else if (isKWAddress(event.to, accounts) && isSWAddress(event.from, accounts)) {
+              buttons = [sendFromKW];
+            } else if (isBitcoinTrx) {
+              if (bitcoinFeatureEnabled) {
+                buttons = isPending ? [viewOnBlockchainButton] : [sendBackBtc];
               } else {
-                buttons = [messageButton, sendBackButtonSecondary];
+                buttons = [];
               }
+            } else if (contactFound) {
+              buttons = isPending
+                ? [messageButton, viewOnBlockchainButtonSecondary]
+                : [messageButton, sendBackButtonSecondary];
             } else if (isPending) {
               buttons = [viewOnBlockchainButton, inviteToPillarButton];
             } else {
               buttons = [sendBackToAddress, inviteToPillarButton];
             }
-          } else if (contactFound) {
-            if (isPending) {
-              buttons = [messageButton, viewOnBlockchainButtonSecondary];
+          } else if (isBitcoinTrx) {
+            if (bitcoinFeatureEnabled) {
```

Comment:
A good check would be bitcoinAddresses.length > 0 for this instead of the feature check

Reply:
Is it, though?
If no address exists, no ActivityFeedItem is visible, so this code will not be used at all.
Now, if we would like to turn off this feature and user has addresses stored, no action buttons will be visible here :)

Reply:
I meant
 ```
else if (isBitcoinTrx) {
              buttons = isPending ? [viewOnBlockchainButton] : [sendBackBtc];
```

Reply:
got the changes here https://github.com/pillarwallet/pillarwallet/commit/629eefb290674cdbf464cadf2a578f9a35f99b87#diff-ec904b6cf8e8afec5137f06ae6e584f5R979